src/callbacks_step/amr.jl

```diff
@@ -124,6 +128,17 @@ function initialize!(cb::DiscreteCallback{Condition,Affect!}, u, t, integrator)
   amr_callback = cb.affect!
   semi = integrator.p
 
+  mesh, _, _, _ = mesh_equations_solver_cache(semi)
+
+  if mesh isa P4estMesh
+    if cb.affect!.p4est_partition_allow_for_coarsening != mesh.p4est_partition_allow_for_coarsening
+      if mpi_isroot()
+        @info "The attribute p4est_partition_allow_for_coarsening from the mesh is changed to `$(cb.affect!.p4est_partition_allow_for_coarsening)`."
+      end
+      mesh.p4est_partition_allow_for_coarsening = cb.affect!.p4est_partition_allow_for_coarsening
+    end
+  end
+
```

Comment:
I have two questions here.
1. This is very minor, but my understanding is that all of this is only relevant for parallel runs. Would it be possible to check `if mesh isa ParallelP4estMesh` instead? That way the info would only be shown when necessary.
2. Did you check that the mesh is definitely repartitioned after this change and before it is possibly coarsened for the first time by the AMR callback? Otherwise the mesh might still be partitioned with `allow_for_coarsening=false` when it is coarsened.

Reply:
Thanks for the review, @lchristm!